summit_blood_samples/blood_sample/upload_views.py

```diff
@@ -237,12 +237,21 @@ def post(self, request, *args, **kwargs):
                 'message': 'Column names not matching'
             })
 
+        # Getting stats of newly uploading file
+        report_ids = BloodSample.objects.values_list('id', flat=True)[
+            ::1]
+        excel_ids = df.Id.values.tolist()
+        
+        # new records are defined as being records with an id greater than the
+        # last id loaded from the CurrentAppointmentBlood report
+        new_records = [idx for idx in excel_ids if idx > max(excel_ids)]
```

Comment:
```suggestion
        new_records = [idx for idx in excel_ids if idx > max(report_ids)]
```
I suspect the above is a typo, but I'm also unsure whether this logic will achieve the desired effect. It depends on whether the Excel file would ever include IDs that fill in 'gaps' within the `BloodSample` report IDs? If that can happen you'll want to keep the definition of `new_records` here as it used to be, and change the test on line 251 below. I'll make a separate suggestion for that.

Reply:
If we do only need the `max(report_ids)` here, then since this is the only use of `report_ids` in the file we're better getting the database to calculate the max for us. So something like
```python
        max_report_ids = BloodSample.objects.all().aggregate(Max('id'))['id__max']
        new_records = [idx for idx in excel_ids if idx > max_report_ids]
```
with
```python
from django.db.models import Max
```
added to the correct place in the imports at the top of the file.